8.0 product_weight migration

florian-dacosta opened the pull request:
Hello,

The goal of this PR is to migrate product_weight and also make some improvements to the module : 
- Calculate gross weight and net weight (instead of net weight only)
- Allow to calculate weight from the product template tree view for multiples templates.

Let me know what you think.
Regards

florian-dacosta committed:
product_weight migration + add gross weight calculation and possibility to calculate weight for multi products

florian-dacosta committed:
fix description + useless code

florian-dacosta committed:
consider product uom and bom uom

florian-dacosta committed:
add readme file,fix pep8

dufresnedavid commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,107 @@
+            domain = [('product_tmpl_id', '=', product_tmpl_id)]
+        else:
+            product_id = self._context and \
+                self._context.get('active_id', False) or False
```
_context is a private variable here, in my opinion, you should avoid using it.

Also, the following expression is redondant:
product_id = self._context and self._context.get('active_id', False) or False
You may change it for
context = self.env.context
...
product_id = context.get('active_id', False)

dufresnedavid commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,107 @@
+                line.product_uom.id,
+                line.product_qty,
+                component_tmpl.uom_id.id)
+            weight_net += (component_tmpl.weight_net or 0.0) * component_qty
```
can be simplified to
weight_net += component_tmpl.weight_net * component_qty

ghost commented:
@florian-dacosta Thanks for your work. Would you mind adding unit tests ?

florian-dacosta committed:
small fixes + add test

florian-dacosta commented:
Thanks for the feedback, I have changed the context issue.
I've also added unitest, but a very simple one, There are so many cases, I am not sure which are needed or not.
If you need others, just explain the case, I will add it.

ghost commented:
:+1: if travis passes the tests

bealdav commented on product_weight/README.rst:
```diff
@@ -0,0 +1,44 @@
+=====
+
+You can update the weight of on product clicking on the button under 
+the weigts. 
```
weight

florian-dacosta replied:
Thanks, I'll fix it!

bealdav commented:
:+1: LGTM

yvaucher commented on product_weight/README.rst:
```diff
@@ -0,0 +1,44 @@
+    :alt: License: AGPL-3
+
+Product Weight Calculation
+===========
```
From http://rst.ninjs.org/

```
System Message: WARNING/2 (<string>, line 5)

Title underline too short.
```

yvaucher commented on product_weight/__openerp__.py:
```diff
@@ -20,24 +21,18 @@
-    'installable': False,
-}
+{'name': 'Product Weight Calculation',
+ 'version': '0.2',
```
This is a migration, Odoo already adds Odoo version before version number. So version bumping is not necessary in case of migration. But ok we can keep it to 0.2

yvaucher commented on product_weight/bom.py:
```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
```
Unnecessary line break

yvaucher commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,116 @@
+                component_tmpl.uom_id.id)
+            weight_net += component_tmpl.weight_net * component_qty
+            weight_gross += component_tmpl.weight * component_qty
+            _logger.warning("%s : %0.2f | %0.2f" % (
```
Why this warning? shouldn't it be info? Or was it for debug?

yvaucher commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,116 @@
+        else:
+            product_ids = context.get('active_ids', [])
+            products = product_obj.browse(product_ids)
+            template_ids = [p.product_tmpl_id.id for p in products]
```
You can use `mapped`

```
products.mapped('product_tmpl_id').ids
```

yvaucher commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,116 @@
+            template_ids = [p.product_tmpl_id.id for p in products]
+        for template_id in template_ids:
+            boms = self.env['mrp.bom'].search(
+                [('product_tmpl_id', '=', template_id)])
```
You are looking for a single bom? You should limit your search with `limit=1` to assign `bom`

bealdav replied:
For my personnal info, is there performance improvement with limit ?

yvaucher commented:
Missing POT file

yvaucher commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,116 @@
+        if not domain_product or not boms:
+            boms = self.env['mrp.bom'].search(domain_template)
+        if boms:
+            bom = boms[0]
```
You won't need this if you limit your search to 1

yvaucher commented on product_weight/wizard/product_weight_update_view.xml:
```diff
@@ -0,0 +1,58 @@
+      <field name="name">Update Product Weight</field>
+      <field name="model">product.weight.update</field>
+      <field name="arch" type="xml">
+	<form string="Update Product Weight" version="8.0">
```
You can remove `version="8.0"` version was only an attribute for transition for new 7.0 web client

yvaucher commented on product_weight/wizard/product_weight_update_view.xml:
```diff
@@ -0,0 +1,58 @@
+      <field name="name">Update Product Weights</field>
+      <field name="model">product.weight.update</field>
+      <field name="arch" type="xml">
+	<form string="Update Product Weight" version="8.0">
```
`version` to remove

yvaucher commented:
Thanks for the migration, see my comments for some nitpicking

florian-dacosta committed:
clean code + add translation

florian-dacosta commented:
Thanks for the review @yvaucher 
I just took your comments into account.

yvaucher commented:
Thanks

:+1: once travis is green

lmignon commented on product_weight/__openerp__.py:
```diff
@@ -20,24 +21,18 @@
+ 'summary': 'Allows to calculate products weight from its components.',
+ 'depends': ['mrp',
+             ],
+ 'demo': [],
```
empty entry is useless

lmignon commented on product_weight/wizard/product_weight_update.py:
```diff
@@ -0,0 +1,115 @@
+            _logger.info("%s : %0.2f | %0.2f" % (
+                bom.product_tmpl_id.name,
+                weight_net, weight_gross))
+        weight_net = weight_net / tmpl_qty
```
``` python
_logger.info("%s : %0.2f | %0.2f",
             bom.product_tmpl_id.name,
             weight_net, weight_gross)
```

lmignon commented:
@florian-dacosta thank you for your work.
Two little comments, otherwise :+1: (once travis is green :smile: )

florian-dacosta committed:
clean code
